scripts/openyurt-deployer/unit_test_workflow.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+echo "PermitRootLogin=yes" | sudo tee -a /etc/ssh/sshd_config
+
+sudo apt-get update -qq
+sudo apt-get install -qq -y openssh-server 
+sudo service ssh start
+eval "$(ssh-agent -s)"
+
+ssh-keygen -t rsa -b 4096 -f ~/.ssh/id_rsa -N "" > /dev/null
+cat > ~/.ssh/config <<EOF
+  Host host.example
+  User runner
+  HostName localhost
+  IdentityFile ~/.ssh/id_rsa
+EOF
+
+# add public key 
+cat - ~/.ssh/id_rsa.pub > ~/.ssh/authorized_keys
+
+sudo service ssh restart
+
+# add private key to ssh agent
+ssh-add ~/.ssh/id_rsa 
+
+chmod og-rw ~/.ssh
+
+go test -timeout 5m
```

Comment:
it would be better to separate the actual test from the setup script into a separate CI workflow step. Then, we would be able to clearly distinguish when setup fails and when the test fails

Reply:
i have tried this numerous time. it does not work out as the ssh permission set up in previous steps would not persist to the next step. 

level=error msg="failed to create network operation not permitted:"